toolkit/tools/scheduler/schedulerutils/implicitprovides.go

```diff
@@ -48,7 +48,7 @@ func InjectMissingImplicitProvides(res *BuildResult, pkgGraph *pkggraph.PkgGraph
 	}
 
 	// Make sure the graph is still a directed acyclic graph (DAG) after manipulating it.
-	err = pkgGraph.MakeDAG()
+	err = pkgGraph.MakeDAG(false, false, nil)
```

Comment:
>err = pkgGraph.MakeDAG(false, false, nil) [](http://example.com/codeflow?start=1&length=41)

Could we split `MakeDAG` into two methods now? The original `MakeDAG`, which doesn't take any arguments and a separate `MakeDAGWithUpstreamAccess` (not married to the name, open to other options). Both methods could call a common, private one inside `pkggraph.go`.

My goal here is to have the name clarify the usage better. The raw booleans and a nil don't help the reader as much. Hiding this complexity in the internal implementation would keep the number of such calls limited.

Reply:
Done.  I split MakeDAG into two in the following way.

Earlier: 
1. `MakeDAG(resolveCyclesFromUpstream, ignoreVersionToResolveSelfDep bool, cloner *rpmrepocloner.RpmRepoCloner)`

Now:
1. `MakeDAGusingUpstreamRepos(resolveCyclesFromUpstream, ignoreVersionToResolveSelfDep bool, cloner *rpmrepocloner.RpmRepoCloner)
`
2. `MakeDAG()` internally calls `MakeDAGusingUpstreamRepos(false, false, nil)`
 
Cannot make `MakeDAGusingUpstreamRepos` function private because grapher calls it with arguments.

Other places (scheduler and pkggraph_test etc) where don't need upstream repos, we call MakeDAG() with no args.

What is achieved with this change: Instead of calling MakeDAG(false, false, nil) a plain MakeDAG() call will be enough in cases where resolution using upstream repos is not required.

Reply:
Thank you! And I like re-using the public `MakeDAGusingUpstreamRepos` instead of my idea with a common private method.:)